morango/models.py

```diff
@@ -301,16 +301,18 @@ class Meta:
         unique_together = ("instance_id", "partition")
 
     @classmethod
-    def update_fsics(cls, fsic1, fsic2, sync_filter):
+    @transaction.atomic
+    def update_fsics(cls, fsics, sync_filter):
+        internal_fsic = DatabaseMaxCounter.calculate_filter_max_counters(sync_filter)
         updated_fsic = {}
-        for key, value in iteritems(fsic1):
-            if key in fsic2:
+        for key, value in iteritems(fsics):
+            if key in internal_fsic:
                 # if same instance id, update fsic with larger value
-                if fsic1[key] > fsic2[key]:
-                    updated_fsic[key] = fsic2[key]
```

Comment:
I think this logic was wrong before, I believe fsic2 should have actually been fsic1.